Week 5/Solution2.java

```diff
@@ -0,0 +1,109 @@
+import java.io.*;
+import java.util.*;
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
```

Comment:
Great job with this + your original implementation :) Thanks for testing and leaving thorough comments!